main.go

```diff
@@ -42,48 +44,101 @@ func main() {
 
 	reader := bufio.NewReader(os.Stdin)
 	for {
-		cmd, err := reader.ReadString('\n')
+		fullCmd, err := reader.ReadString('\n')
 		if err != nil {
-			outputError(err)
+			output(&genericMessageJSON{
+				EventType: "command_error",
+				Error:     true,
+				Message:   err.Error(),
+			})
 			os.Exit(1)
 		}
-		cmd = strings.ToUpper(strings.TrimSpace(cmd))
+		split := strings.Split(fullCmd, " ")
+		cmd := strings.ToUpper(strings.TrimSpace(split[0]))
 		switch cmd {
+		case "HELLO":
+			re := regexp.MustCompile(`(\d+) "([^"]+)"`)
```

Comment:
I think we should avoid compiling this `regexp` for every `HELLO` message by compiling it outside the loop.

Reply:
Moving `regexp.Compile` outside the loop will keep the compiled regexp in RAM forever while keeping it inside the loop will consume CPU cycles to compile the regexp every time the `HELLO` command is executed. Since the `HELLO` command is supposed to be called just once at the beginning, the `regexp.MustCompile` is therefore called just once: moving it outside the loop doesn't give us any advantage (the regexp will be compiled one time in both cases).

BTW the amount of RAM or CPU we are saving in a case or another is negligible, so I will give priority to readability: moving the regexp outside the loop will move it away from his main context (the `HELLO` command handler), reducing code readability.

In conclusion: I will not move it outside the loop :-)

Reply:
Makes sense. :+1: